Bugfix/72 fix develop backend

HijoDeHonor opened the pull request.

HijoDeHonor committed:
fix: resolve endpoint path conflicts and enhance JWT handling

- Changed user endpoint paths to avoid conflicts
- Updated JWT expiration management logic for better security
- Corrected typos and minor grammatical errors
- Implemented CORS to manage cross-origin requests effectively

HijoDeHonor committed:
fix: correct create offers endpoint logic to align with recent changes

- Updated logic of the create offers endpoint to match modifications in other methods
- Ensured consistency and proper functionality across related endpoints

HijoDeHonor committed:
fix: update create offers endpoint and align with frontend

- Added default values for the 'page' prop in getOffers
- Renamed props to align with frontend naming conventions
- Adjusted tests to reflect changes

HijoDeHonor committed:
fix: correct property access and missing props in functions

Corrected property access issues and fixed missing props in various functions. Improved the method for creating database connections.

HijoDeHonor committed:
feat: complete deleteOffer endpoint logic to restore user items

Completed the logic for the deleteOffer endpoint to restore user items upon deletion. Modified the tests accordingly.

PiewDev commented on index.js:
```diff
@@ -1,17 +1,34 @@
-app.disable('x-powered-by');
+app.use(cors({
+  origin: (origin, callback) => {
+    const ALLOWED_ORIGINS = [
```
ALLOWED_ORIGINS must be obtain from .env

PiewDev commented on src/offer/offerRepository.js:
```diff
@@ -81,67 +79,70 @@ export class OfferRepository {
+        WHERE 
         o.id IN (${idsForQuery})
-    `;
+        `;
```
The styles in the written SQL are not standard. See the option to configure the linter to leave it standard or manually unify the styles

PiewDev commented on src/offer/offerService.js:
```diff
@@ -67,18 +79,21 @@ export class OfferService {
+    const { offerItems, userNamePoster } = offer[0];
+
+    const addItemsAndDeleteOffer = await this.offerRepository.addItemsAndDeleteOffer(id, offerItems, userNamePoster);
+    console.log(addItemsAndDeleteOffer);
```
This console log should not go

PiewDev commented on src/offer/offerService.js:
```diff
@@ -67,18 +79,21 @@ export class OfferService {
+    const offer = await this.offerRepository.getOffer(id);
+    const { offerItems, userNamePoster } = offer[0];
+
+    const addItemsAndDeleteOffer = await this.offerRepository.addItemsAndDeleteOffer(id, offerItems, userNamePoster);
```
The name of this variable is not very clear

PiewDev commented on src/offer/offerService.js:
```diff
@@ -26,6 +28,12 @@ export class OfferService {
+    try {
+      await this.inventoryService.removeItemsFromUser(userName, offer);
+    } catch (error) {
+      this.deleteOffer(id);
```
Its correct call service deleteOffer here? because deleteOffer in service add the items to user and here the items never was remove

PiewDev commented on src/utils/mySQL/mySQLConnection.js:
```diff
@@ -7,17 +7,20 @@ export class MySQLConnection {
 
   async executeQuery (query, parameters, connection) {
-    let conn = connection;
+    let conn;
```
Maybe call this like localConnection....

PiewDev commented on src/utils/mySQL/mySQLConnection.js:
```diff
@@ -7,17 +7,20 @@ export class MySQLConnection {
+
     try {
-      if (!conn) {
+      if (!connection) {
```
Maybe call this like externalConnection...

HijoDeHonor committed:
refactor: rename variables for clarity and create new method with updated logic

Renamed variables for better clarity, created a new method to replace an outdated one with changed logic. Moved allowed origins to an environment variable and removed unnecessary console.logs.
